### ISSUE TITLE
FuncUnit 4.0 Prototype

### ISSUE THREAD

**Issue by justinbmeyer** [bitovi/funcunit#253]
- [x] Get a facsimile of all tests to pass again.
- [ ] Clean up the code.  Get rid of all old code, put tests in a nice spot.
- [ ] Make a pre releases.
- [ ] Create remaining FuncUnit methods and test them.  Some funcunit methods that were supported probably don't have tests.  We should implement these.  This training can be useful: https://www.bitovi.com/academy/learn-dom-jquery.html
- [ ] Make sure it works in all browsers.

**Comment by justinbmeyer** [comment]
ACTIONS

- [x] .click()
- [x] .dblclick()
- [x] .drag()
- [x] .move()
- [x] .rightClick()
- [x] .scroll()
- [x] .sendKeys()
- [x] .type()

CSS

.css()

- [x] .hasClass()

DIMENSIONS

- [x] .height()
- [x] .width()
- [x] .innerHeight()
- [x] .innerWidth()
- [x] .outerHeight()
- [x] .outerWidth()
- [x] .scrollLeft()
- [x] .scrollTop()

.offset()
.position()
.size()


MANIPULATION

- [x] .text()
- [x] .val()

.attr()
.html()


TRAVERSAL

- [x] .closest()
- [x] .find()

.first()
.last()
.next()
.prev()
.siblings()


UTILITIES

- [x] useWindow
- [x] F.prompt() (one of these)
- [x] F.confirm()
- [x] F.alert() ? 

F.detach()
F.eval()
F.open()
F.speed()
F.timeout()

WAITS

F.animationEnd()
.missing()

.exists() - not needed
.invisible()

- [x] .visible()

.wait()
